src/rest_framework_reactive/consumers.py

```diff
@@ -29,6 +67,26 @@ class PollObserversConsumer(AsyncConsumer):
         )
 
 
+class ThrottleConsumer(AsyncConsumer):
+    """Consumer for throttling observer evaluations."""
+
+    async def delay_observer_evaluate(self, message):
+        """Throttle observer evaluation."""
+        observer_id = message['observer']
+
+        # Sleep until we need to schedule the next evaluation.
+        await asyncio.sleep(THROTTLE_RATE)
+
+        # Reset throttled flag before scheduling the evaluation.
+        observer_throttled_reset(observer_id)
```

Comment:
You can delete the flag just before calling `WorkerConsumer ._evaluate`, because this only puts the task in the queue, it doesn't guarantee to start it executing right away and it is still ok to discard other events in the meantime. But it will be a bit tricky to differentiate throttled  tasks from the one that are processed for the first time, maybe you can add another key in the channels message.

Reply:
Actually you just remove the flag before the `if` :)